src/main.cpp

```diff
@@ -165,9 +216,16 @@ Usage: irods_http_api [OPTION]... CONFIG_FILE_PATH
 CONFIG_FILE_PATH must point to a file containing a JSON structure containing
 configuration options.
 
+--dump-config can be used to generate a default configuration file. See this
```

Comment:
so, maybe we put the word default or template in the option name...   it won't ever just dump the actual config already in place...

Reply:
Sure. I tried using words like template and wasn't really happy with them. Perhaps seeing it again will be different.

How about ...
- `--dump-template-config`
- `--template-config`

Are there other phrases that work better?

---

I also need to remove the word, default, from the highlighted line.

Reply:
template is good.   default is wrong.  agreed.

how about...  `--dump-config-template`

Reply:
I see your point...

However, definitely hoping for a single-file solution.   JSON should be sufficiently flexible to do what we want in the future.  iRODS server is / has been consolidating to one, for multiple reasons...

`--dump-template config` (TYPE) is not horrible at all.

Reply:
Going with (1)

Reply:
I like the ideas in this conversation. Will link this to the issue directly in case we want to look it up again.

Reply:
done